cmd/frontend/graphqlbackend/repository.go

```
@@ -344,10 +345,38 @@ func (r *RepositoryResolver) Changelist(ctx context.Context, args *RepositoryCha
 		return nil, err
 	}
 
+	// Changelists only exist for perforce repos.
+	if repo.ExternalRepo.ServiceType != extsvc.TypePerforce {
+		return nil, nil
+	}
+
 	rc, err := r.db.RepoCommitsChangelists().GetRepoCommitChangelist(ctx, repo.ID, cid)
 	if err != nil {
+		// Not found could mean either the mapper has some bug, is currently
+		// broken, or hasn't run yet.
+		// We take a slow path here to check if we can find the commit in the repo.
 		if errcode.IsNotFound(err) {
-			return nil, nil
+			r.logger.Debug("failed to find changelist ID - trying to find it via log", log.String("args.CID", args.CID))
+			cs, err := r.gitserverClient.Commits(ctx, r.name, gitserver.CommitsOptions{
+				AllRefs:      true,
+				MessageQuery: fmt.Sprintf("change = %d]", cid),
+				N:            1,
+			})
+			if err != nil {
+				return nil, err
+			}
+			if len(cs) == 0 {
+				return nil, nil
+			}
+			cid, err := perforce.GetP4ChangelistID(string(cs[0].Message))
+			if err != nil {
```

I see that we're swallowing the error here. Can we possibly log it/ add it to the trace?